### PR TITLE
Filter loras by size, add curated lora list

### PR DESCRIPTION
**Pull request by Efreak**
Loras greater than 220mb will be silently(!) ignored by workers. The exceptions to this are in the curated loras list.
This PR
1. downloads the curated Lora list periodically (probably not necessary tbh)
2. filters available loras (shown by autocomplete) by file size or availability in curated lora list
3. When a Lora is requested by ID, checks the file size and curated lora list to ensure it will be processed.

### CODE REVIEW

File: src/classes/client.ts [hunk 1]
```diff
@@ -91,6 +93,14 @@ export class AIHordeClient extends Client {
 		this.horde_style_categories = res
 	}
 
+	async loadHordeCuratedLORAs() {
+		const source = this.config.data_sources?.curated_loras_source ?? `https://raw.githubusercontent.com/Haidra-Org/AI-Horde-image-model-reference/main/lora.json`
```
Comment (ZeldaFan0225) [inline src/classes/client.ts:97]: i would use || instead of ?? in case the config has an empty string

File: src/commands/advanced_generate.ts [hunk 2]
```diff
@@ -660,7 +661,7 @@ ETA: <t:${Math.floor(Date.now()/1000)+(status?.wait_time ?? 0)}:R>`
                 if(!isNaN(Number(option.value)) && option.value) {
                     const lora_by_id = await context.client.fetchLORAByID(option.value, context.client.config.advanced_generate?.user_restrictions?.allow_nsfw)
 
-                    if(lora_by_id?.name) ret.push({
+                    if(lora_by_id?.name && (lora_by_id?.modelVersions[0]?.files[0]?.sizeKB && lora_by_id?.modelVersions[0]?.files[0]?.sizeKB <= 220000 || context.client.horde_curated_loras?.includes(lora_by_id.id))) ret.push({
```
Comment (ZeldaFan0225) [inline src/commands/advanced_generate.ts:664]: i'd add parenthesis around the A || B to make it more clear which ones belong together

**Review by ZeldaFan0225 — CHANGES_REQUESTED** [review]

File: src/classes/client.ts (hunk 1 above)
Comment (Efreak) [reply src/classes/client.ts]: Done. I also updated categories and styles json files to do the same thing.
While I was at it, I noticed that the URLs still default to db0, so I fixed those as well.

File: src/commands/advanced_generate.ts (hunk 2 above)
Comment (Efreak) [reply src/commands/advanced_generate.ts]: Done, but I did it wrong. I was thinking that if it's in the curated list then it doesn't matter if there's a file size, but I think if there's no file size then that means there's no file available (not sure if that's actually possible, tbh) in which case it doesn't matter if it's in the curated list. I'll switch the parentheses around.

Comment (Efreak) [reply src/commands/advanced_generate.ts]: Done

File: src/classes/client.ts [hunk 3]
```diff
@@ -76,21 +78,29 @@ export class AIHordeClient extends Client {
 	}
 
 	async loadHordeStyles() {
-		const source = this.config.data_sources?.styles_source ?? `https://raw.githubusercontent.com/db0/Stable-Horde-Styles/main/styles.json`
+		const source = this.config.data_sources?.styles_source || `https://raw.githubusercontent.com/AI-Horde/Stable-Horde-Styles/main/styles.json`
```
Comment (ZeldaFan0225) [inline src/classes/client.ts:81]: please make sure that the link is actually valid, in this case it is not valid and returns a 404

File: src/classes/client.ts [hunk 4]
```diff
@@ -76,21 +78,29 @@ export class AIHordeClient extends Client {
...
 		const req = await fetch(source)
 		if(!req.status?.toString().startsWith("2")) throw new Error("Unable to fetch styles");
 		const res = await req.json()
 		this.horde_styles = res
 	}
 
 	async loadHordeStyleCategories() {
-		const source = this.config.data_sources?.style_categories_source ?? `https://raw.githubusercontent.com/db0/Stable-Horde-Styles/main/categories.json`
+		const source = this.config.data_sources?.style_categories_source || `https://raw.githubusercontent.com/AI-Horde/Stable-Horde-Styles/main/categories.json`
```
Comment (ZeldaFan0225) [inline src/classes/client.ts:89]: please make sure that the link is actually valid, in this case it is not valid and returns a 404

File: src/commands/horde.ts [hunk 5]
```diff
@@ -23,11 +23,11 @@ export default class extends Command {
...
             title: "AI Horde",
             //TODO: Add more info in the future
-            description: `The [AI Horde](https://https://aihorde.net/) is a crowdsourced service providing Stable Diffusion and other AI services for everyone.\nIt is free, open sourced, and relies on volunteer processing power.\n\nIf you enjoy using it, please consider [onboarding your own PC as a worker](https://github.com/db0/AI-Horde-Worker#readme) or supporting its development [on patreon](https://www.patreon.com/db0)${article ? `\n\n**Latest News**\n${article.newspiece}\n<t:${Math.round(Number(new Date(article.date_published!))/1000)}>` : ""}`
+            description: `The [AI Horde](https://https://aihorde.net/) is a crowdsourced service providing Stable Diffusion and other AI services for everyone.\nIt is free, open sourced, and relies on volunteer processing power.\n\nIf you enjoy using it, please consider [onboarding your own PC as a worker](https://github.com/AI-Horde/AI-Horde-Worker#readme) or supporting its development [on patreon](https://www.patreon.com/db0)${article ? `\n\n**Latest News**\n${article.newspiece}\n<t:${Math.round(Number(new Date(article.date_published!))/1000)}>` : ""}`
```
Comment (ZeldaFan0225) [inline src/commands/horde.ts:26]: please make sure that the link is actually valid, in this case it is not valid and returns a 404

File: template.config.json [hunk 6]
```diff
@@ -250,7 +250,8 @@
         }
     },
     "data_sources": {
+        "curated_loras_source": "https://raw.githubusercontent.com/Haidra-Org/AI-Horde-image-model-reference/main/lora.json",
         "styles_source": "https://raw.githubusercontent.com/Haidra-Org/AI-Horde-Styles/main/styles.json",
-        "style_categories_source": "https://raw.githubusercontent.com/db0/Stable-Horde-Styles/main/categories.json"
+        "style_categories_source": "https://raw.githubusercontent.com/Haidra-Org/Stable-Horde-Styles/main/categories.json"
```
Comment (ZeldaFan0225) [inline template.config.json:255]: please make sure that the link is actually valid, in this case it is not valid and returns a 404

**Review by ZeldaFan0225 — CHANGES_REQUESTED** [review]

**Review by ZeldaFan0225 — APPROVED** [review]

**Pull request merged**